tracpro/test/cases.py

```diff
@@ -0,0 +1,155 @@
+from tracpro.groups.models import Group, Region
+from tracpro.polls.models import Poll, Question
+
+
+class TracProTest(TestCase):
+    """Base class for all test cases in TracPro."""
+
+    def setUp(self):
+        super(TracProTest, self).setUp()
+        self.clear_cache()
+
+    def clear_cache(self):
+        # we are extra paranoid here and actually hardcode redis to 'localhost'
+        # and '10' Redis 10 is our testing redis db
+        r = redis.StrictRedis(host='localhost', db=10)
+        r.flushdb()
+
+    def create_org(self, name, timezone, subdomain):
+        org = Org.objects.create(
+            name=name, timezone=timezone, subdomain=subdomain,
+            api_token=str(uuid4()), created_by=self.superuser,
+            modified_by=self.superuser)
+        org.set_config('facility_code_field', 'facility_code')
+        return org
+
+    def create_region(self, org, name, uuid):
+        return Region.create(org, name, uuid)
+
+    def create_group(self, org, name, uuid):
+        return Group.create(org, name, uuid)
+
+    def create_admin(self, org, full_name, email):
+        user = User.create(
+            None, full_name, email, password=email, change_password=False)
+        user.org_admins.add(org)
+        return user
+
+    def create_user(self, org, full_name, email, regions):
+        return User.create(
+            org, full_name, email, password=email, change_password=False,
+            regions=regions)
+
+    def create_contact(self, org, name, urn, region, group, uuid):
+        user = org.administrators.first()
+        return Contact.create(
+            org, user, name, urn, region, group, 'FC123', 'eng', uuid)
+
+    def login(self, user):
+        result = self.client.login(username=user.username, password=user.username)
+        self.assertTrue(
+            result, "Couldn't login as %(user)s / %(user)s" % dict(user=user.username))
+
+    def switch_region(self, region):
+        session = self.client.session
+        session['region'] = region.pk
+        session.save()
+
+    def url_get(self, subdomain, url, params=None, **extra):
+        if subdomain:
+            extra.setdefault('HTTP_HOST', "{}.testserver".format(subdomain))
+        return self.client.get(url, params, **extra)
+
+    def url_post(self, subdomain, url, data=None, **extra):
+        if subdomain:
+            extra.setdefault('HTTP_HOST', "{}.testserver".format(subdomain))
+        return self.client.post(url, data, **extra)
+
+    def datetime(self, year, month, day, hour=0, minute=0, second=0,
+                 microsecond=0, tz=pytz.UTC):
+        return datetime.datetime(year, month, day,
+                                 hour, minute, second, microsecond, tz)
+
+    def assertLoginRedirect(self, response, subdomain, next):
+        self.assertRedirects(
+            response,
+            'http://%s.testserver/users/login/?next=%s' % (subdomain, next))
+
+
+class TracProDataTest(TracProTest):
+    """Common data set-up."""
```

Comment:
Splitting this out makes sense. I extend TracProTest a few places, so I'll have to start using this new class.